src/library_nodefs.js

```diff
@@ -276,14 +276,22 @@ addToLibrary({
         // Node.js < 6 compatibility: node errors on 0 length reads
         if (length === 0) return 0;
         try {
-          return fs.readSync(stream.nfd, new Int8Array(buffer.buffer, offset, length), { position: position });
+#if MIN_NODE_VERSION < 131300
+          return fs.readSync(stream.nfd, new Int8Array(buffer.buffer, offset, length), 0, length, position);
```

Comment:
Do newer versions of node also support this old pattern?  If so, can we not just use this older pattern everwhere?  Is there any downside?  It looks like it creates less gabage since you don't need to create a new option object each call.

If newer versions don't support this pattern then I suppose we would want a runtime check rather than a compile time one.

Reply:
Newer versions of Node.js ought to support this too. I guarded this for code size reasons, but perhaps that is negligible.

Reply:
Commit 65a6ff350972343da0cf6b2a0d72e1e2af6707e4 prefers the old API instead.